src/nuklear_property.c

```diff
@@ -349,10 +347,10 @@ nk_property(struct nk_context *ctx, const char *name, struct nk_property_variant
     if (!s) return;
 
     /* calculate hash from name */
-    if (name[0] == '#') {
-        hash = nk_murmur_hash(name, (int)nk_strlen(name), win->property.seq++);
-        name++; /* special number hash */
-    } else hash = nk_murmur_hash(name, (int)nk_strlen(name), 42);
+    if (name.len > 0 && name.ptr[0] == '#') {
+        hash = nk_murmur_hash(name, win->property.seq++);
+        name = nk_substr(name, 1, name.len); /* special number hash */
```

Comment:
One thing we could consider is to just have `nk_substr` as a `static` function in the header only. Since it is really small, it shouldn't bloat users code and making it `static` allows the optimizer to inline it if it makes sense.

Reply:
static?  Don't you mean inline or macro?  The static keyword makes a function local to a TU ... I don't think I've ever seen it in a header.  *googles*

Dang inline is even more complicated in C than i realized.  I was only aware of the inline definition in the header, extern inline declaration in a C file version.  Didn't know about static inline.

For those who were unaware like I was, [this link](https://stackoverflow.com/questions/216510/what-does-extern-inline-do/216546#216546) sums up the various versions and (in)compatibilities.

Though, from that link, it seems there's a minor problem.  Nuklear proper is C89, (not GNU89?) which has no defined semantics.  It's probably not a big deal since pretty much no one is using it from strict C89 though.

I do think it's confusing to refer to "making substr static" when static by itself doesn't make any sense in this context.  Saying inline or static inline would be more accurate and clear imo.